### ISSUE TITLE
Agent instrumentation errors with JDBC PreparedStatement

### ISSUE THREAD

**Issue by jhsheets** [Microsoft/ApplicationInsights-Java#277]
Here's the stacktrace:

```
org.jboss.weld.interceptor.proxy.InterceptorException: java.lang.VerifyError: StackMapTable format error: bad type array size in method com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.setTime(ILjava/sql/Time;)V
    at org.jboss.weld.interceptor.proxy.InterceptorInvocationContext.proceed(InterceptorInvocationContext.java:150) ~[weld-servlet.jar:2013-07-17 09:29]
    at org.mybatis.cdi.LocalTransactionInterceptor.invoke(LocalTransactionInterceptor.java:58) ~[mybatis-cdi-1.0.0-beta1.jar:1.0.0-beta1]
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_79]
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_79]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_79]
    at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_79]
    at org.jboss.weld.interceptor.proxy.SimpleMethodInvocation.invoke(SimpleMethodInvocation.java:30) ~[weld-servlet.jar:2013-07-17 09:29]
    at org.jboss.weld.interceptor.chain.AbstractInterceptionChain.invokeNext(AbstractInterceptionChain.java:93) ~[weld-servlet.jar:2013-07-17 09:29]
    at org.jboss.weld.interceptor.chain.AbstractInterceptionChain.invokeNextInterceptor(AbstractInterceptionChain.java:78) ~[weld-servlet.jar:2013-07-17 09:2
```

**Comment by gupele** [comment]
Hi,

Thank you for your input

The method signature has the index: "(ILjava/sql/Time;)V", (the 'I') which makes it difficult to understand. 
Is there a way in which you can narrow down the scenario?

**Comment by jhsheets** [comment]
I noticed that the signature looks correct. I don't know much about
bytecode manipulation to know exactly what the issue is. There is an
overloaded setTime method that takes a Calendar object that you're not
instrumenting. Maybe that's the issue?

**Comment by sfthomas** [comment]
I am facing the same issue - trying to use the SQLServer JDBC4 drivers (version 4.0) in Tomcat, with Hibernate, but it's erroring out with the above. Tried switching to version 6.0, as well as switcing to JDBC4.2, but none of them solved the problem.

**Comment by Dmitry-Matveev** [comment]
As a temporary workaround I would recommend to disable JDBC instrumentation in applicationinsights.xml config to avoid failures if they continue even with version 1.0.7.

**Comment by dhaval24** [comment]
fixed in #449
